Fix terminal TODO: Implement Windows system info display

Oblivionsage opened the pull request:
- Replace hardcoded 'N/A' values with actual Windows system information
- Add GetSystemInfo() for processor architecture detection
- Add GetVersionEx() for Windows version information
- Support both machine-readable and human-readable output formats
- Follow existing Linux uname() implementation pattern
- Maintain cross-platform compatibility

Resolves TODO comment in src/terminal.c line 1257

Oblivionsage committed:
Fix terminal TODO: Implement Windows system info display

- Replace hardcoded 'N/A' values with actual Windows system information
- Add GetSystemInfo() for processor architecture detection
- Add GetVersionEx() for Windows version information
- Support both machine-readable and human-readable output formats
- Follow existing Linux uname() implementation pattern
- Maintain cross-platform compatibility

Resolves TODO comment in src/terminal.c line 1257

ventaquil commented on src/terminal.c:
```diff
@@ -1254,21 +1254,60 @@ void backend_info (hashcat_ctx_t *hashcat_ctx)
       event_log_info (hashcat_ctx, "OS.Name......: Windows");
-      event_log_info (hashcat_ctx, "OS.Release...: N/A");
-      event_log_info (hashcat_ctx, "HW.Platform..: N/A");
+      event_log_info (hashcat_ctx, "OS.Release...: %s", rc_version ? release_buf : "N/A");
```
Is this really necessary? Since you fill `release_buf` with `N/A` maybe this check is pointless.

Oblivionsage replied:
Thanks for the feedback @ventaquil you're right since `release_buf` is already initialized with "N/A", the ternary check is indeed pointless

Should I update it to simply use `release_buf` directly like this?

```c
event_log_info (hashcat_ctx, "OS.Release...: %s", release_buf);
```

And same for the JSON output:

```c
printf ("\"Release\": \"%s\" }, ", release_buf);
```

I appreciate you taking the time to review the code let me know if you'd like me to push an update with this fix

Oblivionsage committed:
Address review feedback: Remove redundant ternary checks

- Remove redundant 'rc_version ? release_buf : "N/A"' checks
- Use release_buf directly since it's already initialized with "N/A"
- Addresses feedback from @ventaquil in PR review

Oblivionsage commented:
I have  updated the code to remove the redundant ternary checks as suggested the fix has been pushed and the build still passes thanks for taking the time to review @ventaquil

ventaquil commented:
You're welcome!
